### ISSUE TITLE
Larq loaded model lost all accuracy, metrics are not valid anymore

### ISSUE THREAD

**Issue by ibarrond** [larq/larq#571]
### Describe the bug
Following the [MNIST tutorial](https://docs.larq.dev/larq/tutorials/mnist/) I trained a BNN, obtaining a respectable accuracy of 97.8%. After saving the model with `model.save("myModel.h5")` and loading the model again with `loadedModel = tf.keras.models.load_model("myModel.h5")`, the accuracy drops to 10% (random choice).

### To Reproduce

```python
#
# Include all code from https://docs.larq.dev/larq/tutorials/mnist/
# [...]
print(f"Test accuracy {test_acc * 100:.2f} %")
#> Test accuracy 97.81 %

model.save("myModel.h5")

loadedModel = tf.keras.models.load_model("myModel.h5")
test_loss, test_acc = loadedModel.evaluate(test_images, test_labels)
print(f"Test accuracy {test_acc * 100:.2f} %")
#> Test accuracy 9.93%
```
### Expected behavior
I expect the test accuracy to be exactly the same as before, 97.81%

### Environment

TensorFlow version: 2.3.0
Larq version: 0.10.0

**Comment by ibarrond** [comment]
Investigating a bit further, I found out that you need to run again `.compile(...)` in order to evaluate correctly.
```python
loadedModel.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])
```
Although all the information seems to be the same in both models (optimizer in `model.optimizer`, loss in `model.compiled_loss`, accuracy metrics in `model.compiled_metrics`), we still need to compile the loaded model with the `loss` and `metrics` in order to evaluate the loaded model.

Shouldn't the saved keras model `"myModel.h5"` contain everything required to load and evaluate the model?

**Comment by leonoverweel** [comment]
Heyy, thanks for checking out Larq and thanks for raising + looking into this!

I found this in the [TF Keras `load_model()` docs on the .h5 format](https://www.tensorflow.org/guide/keras/save_and_serialize#keras_h5_format):

> External losses & metrics added via model.add_loss() & model.add_metric() are not saved (unlike SavedModel). If you have such losses & metrics on your model and you want to resume training, you need to add these losses back yourself after loading the model. Note that this does not apply to losses/metrics created inside layers via self.add_loss() & self.add_metric(). As long as the layer gets loaded, these losses & metrics are kept, since they are part of the call method of the layer.

Assuming that the accuracy here qualifies as an "external loss," it looks like the metrics (even though they are in the original model) don't get saved in the .h5 file, explaining why you needed to re-add them when you reloaded the model. So I think this may be a TF Keras quirk that's not specific to Larq.

One thing you could try, which we also use for some things internally, is using the `SavedModel` format (automatically used if you don't end the name in `model.save("name")` in ".h5"), which *should* keep these metrics. Hope this helps!

**Comment by lgeiger** [comment]
@ibarrond I just checked and this seems to be a bug in TensorFlow itself. Your code example works as expected in TensorFlow 2.2 but breaks in 2.3 even without using any larq layers in the network.

Fortunately this seems to be fixed in the latest TensorFlow nightly, so for the time beeing I recommend to downgrade to TensorFlow 2.2 or use tf-nightly until TensorFlow 2.4 will be released.
